solr/core/src/java/org/apache/solr/store/blob/process/CorePullTask.java

```diff
@@ -354,10 +353,10 @@ void pullCoreFromBlob(boolean isLeaderPulling) throws InterruptedException {
             }
           }
         }
+        concurrencyController.recordState(pullCoreInfo.getCollectionName(), pullCoreInfo.getShardName(), pullCoreInfo.getCoreName(), SharedCoreStage.BLOB_PULL_FINISHED);
```

Comment:
This is meant to be matching BLOB_PULL_STARTED (line#248). At this place, depending on failures, it can be logged without BLOB_PULL_STARTED. You would need to put a new try after line#248 and a finally before line# 320. And then put this line inside that new finally.

Side note: This method needs some refactoring to simplify its flow.